src/middleware/auth.middleware.ts

```diff
@@ -0,0 +1,49 @@
+import passport from 'passport';
+import passportLocal from 'passport-local';
+import passportJwt from 'passport-jwt';
+
+const LocalStrategy = passportLocal.Strategy;
+const JwtStrategy = passportJwt.Strategy;
+const ExtractJwt = passportJwt.ExtractJwt;
+
+import User from '../models/user.model';
+import config from '../config';
+import { NextFunction } from 'express';
+
+export const local = passport.use(new LocalStrategy(User.authenticate()));
+passport.serializeUser(User.serializeUser());
+passport.deserializeUser(User.deserializeUser());
+
+const opts = {
+  jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+  secretOrKey: config.secretKey
+};
+
+export const jwtPasport = passport.use(
+  new JwtStrategy(opts, (jwt_payload: any, done: any) => {
+    User.findOne({ _id: jwt_payload.user }, (err, user) => {
+      if (err) {
+        return done(err, false);
+      }
+      if (user) {
+        return done(null, user);
+      }
+      return done(null, false);
+    });
+  })
+);
+
+export const verifyUser = passport.authenticate('jwt', { session: false });
```

Comment:
Since the JWT doesn't have any user information apart from the passport "id", does it do a database lookup to verify each time?

Reply:
Yes, it does a DB lookup everytime

Reply:
Hmmm. That's an unnecessary overhead. Any way around this?

Reply:
I will have to look into this. I came up with a few solutions but most of them seem to have some or the other side effect. 
Think of an ideal scenario where we have established a system where DB lookup isn't needed. Now I have logged in from 2 systems s1 and s2. If I were to change my password from s2 and then try to access my account from s1..there has to be a DB lookup to invalidate those credentials right? This would contradict the first assumption.

Reply:
Yes, this issue won't arise once we have the refresh token mechanism. Let's add that as an issue and work on it later!